cluster-autoscaler/cloudprovider/mcm/mcm_cloud_provider.go

```diff
@@ -195,6 +193,17 @@ func (mcm *mcmCloudProvider) GetResourceLimiter() (*cloudprovider.ResourceLimite
 // Refresh is called before every main loop and can be used to dynamically update cloud provider state.
 // In particular the list of node groups returned by NodeGroups can change as a result of CloudProvider.Refresh().
 func (mcm *mcmCloudProvider) Refresh() error {
+
+	namespace := mcm.mcmManager.namespace
+	deployment, err := mcm.mcmManager.deploymentLister.Deployments(namespace).Get("machine-controller-manager")
+	if err != nil {
+		return fmt.Errorf("failed to get machine-controller-manager deployment: %v", err.Error())
+	}
+
+	if !(deployment.Status.AvailableReplicas >= 1) {
+		return fmt.Errorf("machine-controller-manager is offline. Cluster autoscaler operations would be suspended.")
+	}
```

Comment:
Can you put this part in a separate method and call it `checkMCMAvailableReplicas`?